File: moviejava/app/src/main/java/com/example/moviejava/data/local/dao/AppDao.java
package com.example.moviejava.data.local.dao;

import static androidx.room.OnConflictStrategy.REPLACE;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;


import com.example.moviejava.data.local.entities.MovieEntity;

import java.util.List;

@Dao
public interface AppDao {
    @Insert(onConflict = REPLACE)
    void addRecord(MovieEntity m);

    @Delete
    void deleteRecord(MovieEntity m);

    @Query("SELECT * FROM movie")
    LiveData<List<MovieEntity>> selectRecord();
}

File: movie/app/src/main/java/com/example/movie/data/local/dao/AppDao.kt
package com.example.movie.data.local.dao

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy.REPLACE
import androidx.room.Query
import com.example.movie.data.local.entities.MovieEntity

@Dao
interface AppDao {
    @Insert(onConflict = REPLACE)
    suspend fun addRecord(m: MovieEntity)

    @Delete
    suspend fun deleteRecord(m: MovieEntity)

    @Query("SELECT * FROM movie")
    fun selectRecord(): LiveData<List<MovieEntity>>
}